mibidata/mibi_image.py

```diff
@@ -199,6 +214,37 @@ def _set_channels(self, channels, length):
                 'Channels must be a list of tuples of (int, str) or a '
                 'list of str')
 
+    def _convert_from_previous_metadata_versions(self):
+        """Convert old metadata format to new one.
+
+        This function ensures backwards compatibility for previous versions.
+        """
+        if self.version != MIBITIFF_VERSION:
+            print("WARNING! you are trying to use the old metadata "
+                  "format for setting the point name. If you are "
+                  "creating a new image, we recommend storing the "
+                  "point name as in the following example: "
+                  "'fov_id': 'Point1', 'fov_name': 'R1C3_Tonsil'.")
```

Comment:
If you are going to emit a warning, use `warnings.warn` not print.

Also, this one doesn't seem quite right -- it checks the version string only, states a recommendation, and then goes ahead and attempt to use that recommendation without confirming that there is a `'point_name'` attribute. Instead, you should probably check for `'point_name'` and if it's present and the version is old, state in the warning that it is being converted.

Reply:
Ok, I restructured the function and updated the warning. In general, I think we should enforce the version check, rather than relying on a check of possible attributes, especially since we are allowing the user to define his own attributes.